diff --git a/node_modules/@oclif/plugin-plugins/lib/plugins.js b/node_modules/@oclif/plugin-plugins/lib/plugins.js
index a96fcd2..8933fdf 100644
--- a/node_modules/@oclif/plugin-plugins/lib/plugins.js
+++ b/node_modules/@oclif/plugin-plugins/lib/plugins.js
@@ -212,9 +212,10 @@ class Plugins {
         if (name.includes('@'))
             return;
         const scope = this.config.pjson.oclif.scope;
+        const prefix = this.config.pjson.oclif.prefix ?? 'plugin';
         if (!scope)
             return;
-        return `@${scope}/plugin-${name}`;
+        return `@${scope}/${prefix}-${name}`;
     }
     async maybeUnfriendlyName(name) {
         const unfriendly = this.unfriendlyName(name);
@@ -227,9 +228,10 @@ class Plugins {
     }
     friendlyName(name) {
         const scope = this.config.pjson.oclif.scope;
+        const prefix = this.config.pjson.oclif.prefix ?? 'plugin';
         if (!scope)
             return name;
-        const match = name.match(`@${scope}/plugin-(.+)`);
+        const match = name.match(`@${scope}/${prefix}-(.+)`);
         if (!match)
             return name;
         return match[1];
