diff --git a/node_modules/oclif/lib/tarballs/build.js b/node_modules/oclif/lib/tarballs/build.js
index ffbbd21..c2a1529 100644
--- a/node_modules/oclif/lib/tarballs/build.js
+++ b/node_modules/oclif/lib/tarballs/build.js
@@ -3,7 +3,9 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.build = void 0;
 const findYarnWorkspaceRoot = require("find-yarn-workspace-root");
 const path = require("path");
+const fs = require("fs")
 const qq = require("qqjs");
+const Arborist = require('@npmcli/arborist');
 const log_1 = require("../log");
 const bin_1 = require("./bin");
 const node_1 = require("./node");
@@ -48,18 +50,28 @@ async function build(c, options = {}) {
     };
     const addDependencies = async () => {
         qq.cd(c.workspace());
-        const yarnRoot = findYarnWorkspaceRoot(c.root) || c.root;
-        const yarn = await qq.exists([yarnRoot, 'yarn.lock']);
+        const workspaceRoot = findYarnWorkspaceRoot(c.root) || c.root;
+        const yarn = await qq.exists([workspaceRoot, 'yarn.lock']);
         if (yarn) {
-            await qq.cp([yarnRoot, 'yarn.lock'], '.');
+            await qq.cp([workspaceRoot, 'yarn.lock'], '.');
             await qq.x('yarn --no-progress --production --non-interactive');
         }
         else {
-            let lockpath = qq.join(c.root, 'package-lock.json');
-            if (!await qq.exists(lockpath)) {
-                lockpath = qq.join(c.root, 'npm-shrinkwrap.json');
+            let lockpath = qq.join(workspaceRoot, 'package-lock.json');
+            if (!(await qq.exists(lockpath))) {
+              lockpath = qq.join(workspaceRoot, 'npm-shrinkwrap.json');
             }
             await qq.cp(lockpath, '.');
+            const arb = new Arborist();
+            await arb.loadVirtual();
+            await arb.buildIdealTree({
+              update: {
+                names: Array.from(arb.virtualTree.fsChildren)
+                  .filter((node) => !node.dev && node !== arb.virtualTree.root)
+                  .map((node) => node.packageName),
+              }
+            });
+            await arb.reify();
             await qq.x('npm install --production');
         }
     };
